airbyte-cdk/python/airbyte_cdk/sources/declarative/date/min_max_date.py

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+
+
+class MinMaxDate:
+    """
+    Compares the provided date against optional minimum or maximum times. If date is earlier than
+    min_date, then min_date is returned. If date is greater than max_date, then max_date is returned.
+    """
+
+    def __init__(
+        self,
+        date: str,
+        datetime_format: str = "%Y-%m-%dT%H:%M:%S.%f%z",
+        min_date: str = "",
+        max_date: str = "",
+    ):
+        self._date_interpolator = InterpolatedString(date)
+        self._datetime_format = datetime_format
+        self._timezone = datetime.timezone.utc
```

Comment:
Follows the same pattern as the DatetimeStreamSlicer that defaults to UTC. If we decide this needs configurability, it can ideally follow the same pattern and inherit it from the DatetimeStreamSlicer's options